library/src/main/java/com/novoda/downloadmanager/MigrationStatusNotificationCreator.java

```diff
@@ -29,7 +29,7 @@ public NotificationInformation createNotification(final MigrationStatus migratio
         return new NotificationInformation() {
             @Override
             public int getId() {
-                return migrationStatus.hashCode();
+                return migrationStatus.migrationId().hashCode();
```

Comment:
Now the `NotificationDispatcher` can correctly switch between the different types of notifications 😄 if it was left how it was then it would keep the persisted notification **and** display the completed notification.